### PR TITLE
Create tab-panel-behavior

### PR DESCRIPTION
**Pull request by svanherk**
Create `d2l-tab-panel-behavior` so we can create custom `tab-panel`s and can put `d2l-tabs` in a separate component.  I added a demo to simulate the scenario I need for the filter UI.

### CODE REVIEW

File: d2l-tabs.js [hunk 1]
```diff
@@ -716,19 +716,30 @@ Polymer({
...
+		}
+	},
+
 	_handleTabTextUpdate: function(e) {
 		var panel = dom(e).rootTarget;
 		var tab = dom(this.root).querySelector(`[aria-controls='${panel.id}']`);
 
 		fastdom.mutate(function() {
 			tab.text = e.detail.text;
+			this._handleResize();
```
Comment (svanherk) [inline d2l-tabs.js]: This is an unrelated bug I noticed.  If the title gets longer, sometimes it can go offscreen and we should trigger a resize to show the arrows instead.

File: d2l-tabs.js [hunk 2]
```diff
@@ -499,7 +499,7 @@ Polymer({
 
 			for (var i = 0; i < info.removedNodes.length; i++) {
 				var removedNode = info.removedNodes[i];
-				if (removedNode.tagName === 'D2L-TAB-PANEL' && removedNode.id.length > 0) {
+				if (removedNode.nodeType === 1 && removedNode.getAttribute('role') === 'tabpanel' && removedNode.id.length > 0) {
```
Comment (dbatiste) [inline d2l-tabs.js]: Super mind thing: if preferred, I think it would be possible to use the browser constant `Node.ELEMENT_NODE`.  We know `1` refers to an element, but some folks might not.   I would not go out of my way to change though.

File: d2l-tabs.js (hunk 1 above)
Comment (dbatiste) [reply d2l-tabs.js]: Good find!

File: d2l-tabs.js (hunk 2 above)
Comment (svanherk) [reply d2l-tabs.js]: I do like that better, it reads a lot nicer.  Will update
